### PR TITLE
Node package updates

### PR DESCRIPTION
**Pull request by stephenfuqua**
These updates significantly reduce the number of vulnerabilities reported by `npm audit report`. However, some vulnerabilities could not be fixed without potentially breaking changes. Updating from React 4 to React 5 will likely fix most of the remaining problems, except for `faker`.

### CODE REVIEW

File: src/API/LeadershipProfileAPI/ClientApp/package-lock.json
Comment (stephenfuqua) [inline src/API/LeadershipProfileAPI/ClientApp/package-lock.json:1]: `package-lock.json` should always be kept in source control for repeatable builds.

File: src/API/LeadershipProfileAPI/ClientApp/package.json [hunk 1]
```diff
@@ -21,11 +21,24 @@
     "react-export-excel": "^0.5.3",
     "react-export-table-to-excel": "^1.0.6",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "^4.0.0",
+    "react-scripts": "^4.0.3",
     "reactstrap": "^8.7.1",
     "secure-web-storage": "^1.0.2",
     "web-vitals": "^0.2.4",
-    "xlsx": "^0.18.5"
+    "xlsx": "https://cdn.sheetjs.com/xlsx-0.20.1/xlsx-0.20.1.tgz"
```
Comment (stephenfuqua) [inline src/API/LeadershipProfileAPI/ClientApp/package.json:28]: The maintainer of `xlsx` has a dispute with npmjs.org and stopped posting updates there. The only way to install updates now is by direct link to the tarball.

File: src/API/LeadershipProfileAPI/ClientApp/package.json [hunk 2]
```diff
@@ -5,10 +5,10 @@
   "dependencies": {
     "@testing-library/jest-dom": "^5.11.6",
     "@testing-library/react": "^11.1.2",
-    "@testing-library/user-event": "^12.2.2",    
+    "@testing-library/user-event": "^12.2.2",
     "bootstrap": "^4.5.3",
     "chart.js": "^3.9.1",
-    "crypto-js": "^4.0.0",
+    "crypto-js": "^4.2.0",
     "faker": "^6.6.6",
```
Comment (stephenfuqua) [inline src/API/LeadershipProfileAPI/ClientApp/package.json:12]: This version of faker is no longer supported and should be replaced with `@faker-js/faker`. I don't know if there are breaking changes - did not try it. I'm a little surprised to see a faker library in a front-end application, but I assume it is here for a reason.

File: src/API/LeadershipProfileAPI/ClientApp/package.json [hunk 3]
```diff
@@ -21,11 +21,24 @@
...
+    "semver": "^7.5.2",
+    "nth-check": "^2.0.2",
+    "node-forge": "^1.3.1",
+    "minimatch": "^3.1.2",
+    "loader-utils": "^2.0.4",
+    "jszip": "^3.8.0",
+    "immer": "^9.0.21",
+    "ejs": "^3.1.8",
+    "browserslist": "^4.17.0",
+    "ansi-html": "^0.0.9"
```
Comment (stephenfuqua) [inline src/API/LeadershipProfileAPI/ClientApp/package.json:41]: These are transitive dependencies that had vulnerabilities. `npm run build` still builds cleanly with these updates.

File: src/API/LeadershipProfileAPI/ClientApp/package-lock.json
Comment (nsd-juanignacio) [reply src/API/LeadershipProfileAPI/ClientApp/package-lock.json:1]: Thanks

File: src/API/LeadershipProfileAPI/ClientApp/package.json (hunk 2 above)
Comment (nsd-juanignacio) [reply src/API/LeadershipProfileAPI/ClientApp/package.json:12]: I will look into this.

File: src/API/LeadershipProfileAPI/ClientApp/package.json (hunk 1 above)
Comment (nsd-juanignacio) [reply src/API/LeadershipProfileAPI/ClientApp/package.json:28]: good tip

File: src/API/LeadershipProfileAPI/ClientApp/package.json (hunk 3 above)
Comment (nsd-juanignacio) [reply src/API/LeadershipProfileAPI/ClientApp/package.json:41]: Sure, I can test the build.

**Review by nsd-juanignacio — APPROVED** [review]
LGTM

**Pull request merged**